setup.py

```diff
@@ -10,24 +10,95 @@
 # SPDX-License-Identifier: ISC
 
 import setuptools
+import os
+import re
+import sys
+import subprocess
+
+from distutils.version import LooseVersion
+from setuptools import Extension
+from setuptools.command.build_ext import build_ext
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
+
+# Based on: https://www.benjack.io/2018/02/02/python-cpp-revisited.html
+# GitHub: https://github.com/benjaminjack/python_cpp_example
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir='', prefix=''):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+        self.prefix = prefix
+
+
+class CMakeBuild(build_ext):
+    def run(self):
+        try:
+            try:
+                out = subprocess.check_output(['cmake', '--version'])
+            except OSError:
+                raise RuntimeError(
+                    "CMake must be installed to build "
+                    "the following extensions: " +
+                    ", ".join(e.name for e in self.extensions))
+
+            cmake_version = LooseVersion(
+                re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+            if cmake_version < '3.7.0':
+                raise RuntimeError("CMake >= 3.7.0 is required.")
+
+            for ext in self.extensions:
+                self.build_extension(ext)
+
+        except BaseException:
+            print(
+                "Failed to build ANTLR parser, "
+                "falling back on slower textX parser.")
+
+    def build_extension(self, ext):
+        extdir = os.path.join(
+            os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name))),
+            ext.prefix)
+        cmake_args = [
+            '-DCMAKE_INSTALL_PREFIX=' + extdir,
+            '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
+            '-DPYTHON_EXECUTABLE=' + sys.executable
+        ]
+
+        cfg = 'Debug' if self.debug else 'Release'
+        build_args = ['--config', cfg]
+        cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
+        build_args += ['--', '-j2']
+
+        env = os.environ.copy()
+        env['CXXFLAGS'] = '{} -DVERSION_INFO=\\"{}\\"'.format(
+            env.get('CXXFLAGS', ''), self.distribution.get_version())
+        if not os.path.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+        subprocess.check_call(
+            ['cmake', ext.sourcedir] + cmake_args,
+            cwd=self.build_temp,
+            env=env)
+        subprocess.check_call(
+            ['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
+        subprocess.check_call(['cmake', '--install', '.'], cwd=self.build_temp)
+        subprocess.check_call(['ctest'], cwd=self.build_temp)
+        print()  # Add an empty line for cleaner output
+
+
 setuptools.setup(
     name="fasm",
-    version="0.0.1",
+    version="0.0.2",
```

Comment:
Please create an issue to move towards this being the `git-describe` output value.

Reply:
https://github.com/SymbiFlow/fasm/issues/34